src/cluster_legacy.c

```diff
@@ -2936,6 +2936,10 @@ void clusterProcessPingExtensions(clusterMsg *hdr, clusterLink *link) {
             if (n && n != myself && !(nodeIsReplica(myself) && myself->replicaof == n)) {
                 sds id = sdsnewlen(forgotten_node_ext->name, CLUSTER_NAMELEN);
                 dictEntry *de = dictAddOrFind(server.cluster->nodes_black_list, id);
+                if (dictGetKey(de) != id) {
+                    /* The dict did not take ownership of the id string, so we need to free it. */
+                    sdsfree(id);
+                }
```

Comment:
I don't remember when we introduced this API `dictAddOrFind`. I have been used to performing `dictFind` first and perform an insertion if dict entry is `NULL`.

Reply:
Git blame says 2016:

    09a50d34a2 src/dict.c (antirez           2016-09-14 16:43:38 +0200  654) dictEntry *dictAddOrFind(dict *d, void *key) {